contracts/message/messagebus/MessageBusReceiver.sol

```diff
@@ -416,9 +428,33 @@ contract MessageBusReceiver is Ownable {
         if (ok) {
             return abi.decode((res), (IMessageReceiverApp.ExecuctionStatus));
         }
+        handleExecutionRevert(gasLeftBeforeExecution, res);
         return IMessageReceiverApp.ExecuctionStatus.Fail;
     }
 
+    function handleExecutionRevert(uint256 _gasLeftBeforeExecution, bytes memory _returnData) private {
+        uint256 gasLeftAfterExecution = gasleft();
+        uint256 maxTargetGasLimit = block.gaslimit - preExecuteMessageGasUsage;
+        if (_gasLeftBeforeExecution < maxTargetGasLimit && gasLeftAfterExecution <= _gasLeftBeforeExecution / 64) {
```

Comment:
Add a comment explaining that in this case, the executor must have not provided sufficient gas limit?

Reply:
done